sharding/notary/service_test.go

```diff
@@ -172,3 +179,123 @@ func TestJoinNotaryPool(t *testing.T) {
 	}
 
 }
+
+func TestLeaveNotaryPool(t *testing.T) {
+	backend, smc := setup()
+	node := &smcClient{smc: smc, t: t, depositFlag: true, backend: backend}
+
+	// Test Leaving Notary Pool Before Joining it
+
+	err := leaveNotaryPool(node)
+	backend.Commit()
+
+	if err == nil {
+		t.Error("Able to leave Notary pool despite having not joined it")
+	}
+
+	// Roundtrip Test , Join and leave pool
+
+	err = joinNotaryPool(node)
+	if err != nil {
+		t.Error(err)
+	}
+	backend.Commit()
+
+	err = leaveNotaryPool(node)
+	backend.Commit()
+
+	if err != nil {
+
+		t.Fatal(err)
+	}
+	numNotaries, err := smc.NotaryPoolLength(&bind.CallOpts{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if big.NewInt(0).Cmp(numNotaries) != 0 {
+		t.Fatalf("Unexpected number of notaries. Got %d, wanted 0.", numNotaries)
+	}
+
+}
+
+func TestReleaseNotary(t *testing.T) {
+	backend, smc := setup()
+	node := &smcClient{smc: smc, t: t, depositFlag: true, backend: backend}
+
+	// Test Release Notary Before Joining it
+
+	err := releaseNotary(node)
+	backend.Commit()
+
+	if err == nil {
+		t.Error("Released From Notary despite Never Joining Pool")
+	}
+
+	// Roundtrip Test , Join and leave pool and release Notary
+
+	err = joinNotaryPool(node)
+	if err != nil {
+		t.Error(err)
+	}
+	backend.Commit()
+
+	err = leaveNotaryPool(node)
+	backend.Commit()
+
+	if err != nil {
+
+		t.Fatal(err)
+	}
+
+	// The remaining part of the test is dependant on access to chainreader
+	// Unable to access block number for releaseNotary
+
+	/*
+
+		balance, err := backend.BalanceAt(context.Background(), addr, nil)
+		if err != nil {
+			t.Error("unable to retrieve balance")
+		}
+
+		err = releaseNotary(node)
+		backend.Commit()
+
+		if err != nil {
+
+			t.Fatal(err)
+		}
+
+		nreg, err := smc.NotaryRegistry(&bind.CallOpts{}, addr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if nreg.Deposited {
+			t.Error("Unable to release Notary and deposit money back")
+		}
+
+		newbalance, err := backend.BalanceAt(context.Background(), addr, nil)
+		if err != nil {
+			t.Error("unable to retrieve balance")
+		}
+
+		if balance.Cmp(newbalance) != 1 {
+			t.Errorf("Deposit was not returned, balance is currently:", newbalance)
+		}
+	*/
```

Comment:
What's up with this commented block?

Reply:
I need chainreader to run the test otherwise the test fails. For the notary we need access for the current period when testing so we need to be able to use this:
```
client.ChainReader().BlockByNumber()
```
To be able to get the current block and therefore the number. So I 
commented it out until #164 is implemented